src/storage/mvcc/txn.rs

```diff
@@ -997,6 +997,13 @@ impl<S: Snapshot, P: PdClient + 'static> MvccTxn<S, P> {
                     return Ok((TxnStatus::TtlExpire, released));
                 }
 
+                if !caller_start_ts.is_zero() && lock.use_async_commit {
+                    return Err(ErrorInner::Other(box_err!(
```

Comment:
I perfer not pushing the `min_commit_ts` to returning an error. If the primary lock's `min_commit_ts` is greater than `caller_start_ts`, the caller won't be blocked even if the `min_commit_ts` isn't pushed.

Reply:
But in my understanding, at this time, once `use_async_commit` is true, the client must use `caller_start_ts = 0` to check txn status. (Although it's not updated in TiDB now)

Reply:
I think both of your ideas are ok 🤔 after all we must refuse pushing and tell the client that the commit ts is not pushed in that case

Reply:
OK. I find that not returning the error also helps async commit to work (not completely correctly though) before recovery is implemented in TiDB.

Reply:
I think since the convention stated [here](https://github.com/tikv/tikv/pull/8258) has been approved, it's better to forbid calling check_txn_status with `caller_start_ts` and `current_ts` set.